libraries/AP_GPS/AP_GPS_UBLOX.cpp

```diff
@@ -1020,28 +1020,31 @@ AP_GPS_UBLOX::_parse_gps(void)
             return false;
 #endif
 
-        case MSG_CFG_SBAS:
-            if (gps._sbas_mode != 2) {
+        case MSG_CFG_SBAS: {
+            uint8_t desired_sbas_mode = get_desired_sbas_mode();
+            if (gps._sbas_mode != desired_sbas_mode) {
 	        Debug("Got SBAS settings %u %u %u 0x%x 0x%x\n", 
                       (unsigned)_buffer.sbas.mode,
                       (unsigned)_buffer.sbas.usage,
                       (unsigned)_buffer.sbas.maxSBAS,
                       (unsigned)_buffer.sbas.scanmode2,
                       (unsigned)_buffer.sbas.scanmode1);
-                if (_buffer.sbas.mode != gps._sbas_mode) {
-                    _buffer.sbas.mode = gps._sbas_mode;
+                if (_buffer.sbas.mode != desired_sbas_mode) {
+                    _buffer.sbas.mode = desired_sbas_mode;
                     _send_message(CLASS_CFG, MSG_CFG_SBAS,
                                   &_buffer.sbas,
                                   sizeof(_buffer.sbas));
                     _unconfigured_messages |= CONFIG_SBAS;
                     _cfg_needs_save = true;
+                    gcs().send_text(MAV_SEVERITY_INFO, "UBLOX setting sbas mode(%d)", desired_sbas_mode);
```

Comment:
Could you move this up before `_buffer.sbas.mode = desired_sbas_mode;` on L1033 and print the current mode and desired mode in the gcs message? That will help us easily check whether the mode reset with powercycle

Also could this be MAV_SEVERITY_ALERT?

Reply:
Done!